src/components/Tokens/TokenDetails/PriceChart.tsx

```diff
@@ -258,7 +258,23 @@ export function PriceChart({ width, height, prices: originalPrices, timePeriod }
     setDisplayPrice(endingPrice)
   }, [setCrosshair, setDisplayPrice, endingPrice])
 
+  useEffect(() => {
+    setCrosshair(null)
+  }, [timePeriod])
```

Comment:
Is this accurate?

```suggestion
  // Resets the crosshair when the time period is changed, to avoid stale UI
  useEffect(() => {
    setCrosshair(null)
  }, [timePeriod])
```

Reply:
yeah thats correct! when i was working on the pr i had bugs where the crosshair wouldn't reset so when i switched graph it looked like it was just hovering at a random location on the new graph. i guess i forgot to mention that 😅